make-prs.py

```diff
@@ -0,0 +1,251 @@
+import argparse
+import logging
+import os
+import re
+import shutil
+import subprocess
+import typing
+
+import github
+import requests
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+WORKING_DIR = "./tmp"
+DIFF_FILE = "./diff.patch"
+
+auth = github.Auth.Token(os.getenv("GITHUB_TOKEN"))
+gh_client = github.Github(auth=auth)
+
+
+def get_diff_as_patch(repo: str, pull_request_number: int):
+    logger.info("Getting diff from PR ...")
+
+    pr_diff_url = gh_client.get_repo(repo).get_pull(pull_request_number).diff_url
+
+    res = requests.get(pr_diff_url)
+    if res.status_code != requests.codes.ok:
+        logger.error(f"Error getting diff from PR {pull_request_number}.")
+        exit(1)
+
+    with open(DIFF_FILE, "w") as f:
+        f.write(res.text)
+
+
+def clone(repo: str):
+    logger.info("Cloning repository ...")
+
+    shutil.rmtree(WORKING_DIR, ignore_errors=True)
+
+    clone_url = f"git@github.com:{repo}.git"
+    command = ["git", "clone", clone_url, WORKING_DIR]
```

Comment:
It seems like this would be a good use-case for [git worktree](https://git-scm.com/docs/git-worktree), so that you get a fresh directory but don't need to re-clone. As long as you're running the script in the repo then that would work very easily, I think.

If we stick with `clone` then I think this can be `--depth=1 --no-single-branch` to avoid getting all the histories of each branch.

Reply:
I think it's simpler if we use a clone with a temporary directory so you don't have to switch trees. I will add `--depth=1 --no-single-branch`.